validator/sawtooth_validator/execution/executor.py

```diff
@@ -299,37 +294,24 @@ def _execute_schedule(self):
 
             # Since we have already checked if the transaction should be failed
             # all other cases should either be executed or waited for.
-            self._execute_or_wait_for_processor_type(
+            self._execute(
                 processor_type=processor_type,
                 content=content,
                 signature=txn.header_signature)
 
         self._done = True
 
-    def _execute_or_wait_for_processor_type(
-            self, processor_type, content, signature):
-        processor = self._processors.get_next_of_type(
-            processor_type=processor_type)
-        if processor is None:
-            LOGGER.debug("no transaction processors registered for "
-                         "processor type %s", processor_type)
-            if processor_type not in self._waiters_by_type:
-                in_queue = queue.Queue()
-                in_queue.put_nowait((content, signature))
-                waiter = _Waiter(
-                    self._send_and_process_result,
-                    processor_type=processor_type,
-                    processors=self._processors,
-                    in_queue=in_queue,
-                    waiters_by_type=self._waiters_by_type)
-                self._waiters_by_type[processor_type] = waiter
-                self._waiting_threadpool.submit(waiter.run_in_threadpool)
-            else:
-                self._waiters_by_type[processor_type].add_to_in_queue(
-                    (content, signature))
-        else:
-            connection_id = processor.connection_id
-            self._send_and_process_result(content, connection_id, signature)
+    def _execute(self, processor_type, content, signature):
+        try:
+            processor = self._processors.get_next_of_type(
+                processor_type=processor_type)
+        except WaitCancelledException:
+            LOGGER.exception("Transaction %s cancelled while "
+                             "waiting for available processor",
+                             content.signature)
```

Comment:
This should probably return so that we don't try to execute `_send_and_process_result` without a processor